src/blueprints/tv2/action-factories/tv2-robot-action-factory.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  Tv2Action,
+  Tv2ActionContentType,
+  Tv2ActionSubtype,
+  Tv2PieceAction,
+  Tv2RobotAction
+} from '../value-objects/tv2-action'
+import { PieceActionType } from '../../../model/enums/action-type'
+import { PieceInterface } from '../../../model/entities/piece'
+import { Action, ActionArgumentType, MutateActionMethods, MutateActionType } from '../../../model/entities/action'
+import { Tv2MisconfigurationException } from '../exceptions/tv2-misconfiguration-exception'
+import { Tv2SourceLayer } from '../value-objects/tv2-layers'
+import { PieceLifespan } from '../../../model/enums/piece-lifespan'
+import { TransitionType } from '../../../model/enums/transition-type'
+import { Tv2PieceInterface } from '../entities/tv2-piece-interface'
+import { Tv2PieceType } from '../enums/tv2-piece-type'
+import { Tv2OutputLayer } from '../enums/tv2-output-layer'
+import {
+  Tv2RobotTimelineObjectFactory
+} from '../timeline-object-factories/interfaces/tv2-robot-timeline-object-factory'
+
+export class Tv2RobotActionFactory {
+
+  constructor(private readonly robotTimelineObjectFactory: Tv2RobotTimelineObjectFactory) {
+  }
+
+  public createRobotActions(): Tv2PieceAction[] {
+    return [
+      this.createCallRobotPresetAction()
+    ]
+  }
+
+  private createCallRobotPresetAction(): Tv2RobotAction {
+    return {
+      id: 'callRobotPresetAction',
+      name: 'Robot Preset',
+      description: 'Calls the preset parsed as the argument',
+      type: PieceActionType.INSERT_PIECE_AS_ON_AIR,
+      data: {
+        pieceInterface: {} as PieceInterface
+      },
+      metadata: {
+        contentType: Tv2ActionContentType.ROBOT,
+        actionSubtype: Tv2ActionSubtype.CALL_PRESET
+      },
+      argument: {
+        name: 'Preset',
+        description: 'The preset to be called provided as a number',
+        type: ActionArgumentType.NUMBER
+      }
+    }
+  }
+
+  public isRobotAction(action: Tv2Action): action is Tv2RobotAction {
+    return action.metadata.contentType === Tv2ActionContentType.ROBOT
```

Comment:
In my experience, I personally like to pass the smallest unit of interest to the function that will use it. In that case I would send only `contentType` as parameter, not full action. Is there a reason to do it this way?

Reply:
In the end it comes down to personal preference, but we are writing object oriented programming, so we should utilize that we have objects. So we can parse our object around instead of breaking down our objects.

This gives, in my opinion, the benefit that the data we are parsing around always corresponds to our entities. We don't need to remember which entity they are a part of because we have the entity.

Another thing I have noticed when people try to provide the smallest information as possible is, that often end up with multiple parameters that could just have been a single parameter.